pycti/connector/opencti_connector_helper.py

```diff
@@ -972,19 +1076,256 @@ def get_state(self) -> Optional[Dict]:
+                    return False
+                else:
+                    # Set buffering = True
+                    self.connector_info.buffering = True
+                    return True
+
+            else:
+                self.metric.inc("error_count")
+                self.connector_logger.error(
+                    "[ERROR] An error occurred while retrieving connector details"
+                )
+                sys.excepthook(*sys.exc_info())
+        except Exception as err:
+            self.metric.inc("error_count")
+            self.connector_logger.error(
+                "[ERROR] An error occurred while checking the queue size",
+                {"reason": str(err)},
+            )
+            sys.excepthook(*sys.exc_info())
+
+    def schedule_unit(
+        self,
+        message_callback: Callable[[], None],
+        duration_period: Union[int, float, str],
+        time_unit: TimeUnit,
+    ) -> None:
+        """
+        This (deprecated) method is there to manage backward compatibility of intervals on connectors,
+        allows you to calculate the duration period of connectors in seconds with time_unit and will be
+        replaced by the "schedule_iso" method. It uses a TimeUnit enum.
+
+        :param message_callback: Corresponds to the connector process
+        :param duration_period: Corresponds to the connector interval, it can vary depending on the connector
+        configuration.
+        :param time_unit: The unit of time for the duration_period.
+        Enum TimeUnit Valid (YEARS, WEEKS, DAYS, HOURS, MINUTES, SECONDS)
+        :return: None
+        """
+        try:
+            # Calculates the duration period in seconds
+            time_unit_in_seconds = time_unit.value[0]
+            duration_period_in_seconds = int(
+                float(duration_period) * time_unit_in_seconds
+            )
+
+            # Start schedule_process
+            self.schedule_process(message_callback, duration_period_in_seconds)
+
+        except Exception as err:
+            self.metric.inc("error_count")
+            self.connector_logger.error(
+                "[ERROR] An unexpected error occurred during schedule_unit",
+                {"reason": str(err)},
+            )
+            sys.excepthook(*sys.exc_info())
+
+    def schedule_iso(
+        self, message_callback: Callable[[], None], duration_period: str
+    ) -> None:
+        """
+        This method allows you to calculate the duration period of connectors in seconds from ISO 8601 format
+        and start the scheduler process.
+
+        :param message_callback: Corresponds to the connector process
+        :param duration_period: Corresponds to a string in ISO 8601 format "P18Y9W4DT11H9M8S"
+        :return: None
+        """
+        try:
+            # Calculates the duration period in seconds
+            timedelta_adapter = TypeAdapter(datetime.timedelta)
+            td = timedelta_adapter.validate_python(duration_period)
```

Comment:
Changing the duration_period from `"PT5S"` to `"0"` raised an error

```
  duration_period: '0'
``` 
![image](https://github.com/OpenCTI-Platform/client-python/assets/75518128/2996fea9-176a-44d0-8b48-0275dea40ab3)

Doing this works 

```suggestion
            td = timedelta_adapter.validate_python(int(duration_period))
```

BUT if we doing this, cannot handle the value in ISO8601 format anymore:

![image](https://github.com/OpenCTI-Platform/client-python/assets/75518128/d7efc3a2-7b6e-48f6-b8be-4cbb9c23985d)

But the value of duration_period in int works if it's 0 

```
  duration_period: 0
``` 

Let's talk about it :)